datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -766,7 +766,7 @@ macro_rules! get_data_page_statistics {
                         [<$stat_type_prefix Int32DataPageStatsIterator>]::new($iterator)
                             .map(|x| {
                                 x.into_iter().filter_map(|x| {
```

Comment:
Changing these made the tests pass, but it seems potentially incorrect to me? It is the same thing as is being done in the row summary tests. See https://github.com/apache/datafusion/blob/3421b52605b00cd2e5a6498ea210cce196a19496/datafusion/core/src/datasource/physical_plan/parquet/statistics.rs#L331-L336

Reply:
Ok - I think I've figured it out. This is actually fixing a bug because the Int32 or Int64 *should* be coerced into an unsigned value.

Reply:
I think the parquet physical types are always signed (because of its heavy java influence) so when storing unsigned values in arrow they need to be cast using `as`